### PR TITLE
OBPIH-2512 Dashboard: Imminent expiry

### PR DESCRIPTION
**Pull request by bamoure**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy [hunk 1]
```diff
@@ -11,20 +11,47 @@ import org.pih.warehouse.tablero.IndicatorDatasets
...
+    def dashboardService
     Date today = new Date()
 
-    DataGraph getExpirationSummaryData(def expirationData) {
-        List listData = []
-        for(item in expirationData){
-            def tmp = item.value? item.value : 0
-            listData.push(tmp)
-        }
+    DataGraph getExpirationSummaryData(Location location, def params) {
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: @augusto-moraes Could you add some comments to explain what's going on here?

File: src/js/components/tablero/GraphCard.jsx [hunk 2]
```diff
@@ -17,9 +17,12 @@ const GraphCard = SortableElement(({
   const cardData = data;
   let graph;
   let filter = 0;
+  let label = 'Last';
```
Comment (awalkowiak) [inline src/js/components/tablero/GraphCard.jsx]: What's the point of those changes?

File: grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy [hunk 3]
```diff
@@ -11,20 +11,47 @@ import org.pih.warehouse.tablero.IndicatorDatasets
...
         
+        Date tempDate = today.clone()
+
+        expirationAlerts.each {
+            Integer daysCounter = 0
+            if(it.inventoryItem.expires != "never") {
+                if (it.daysToExpiry > 0) {
+                    listLabels = []
+                    for (int i=0; i<=querySize; i++) {
+                        tempDate.set(month: today.month + i, date: i ? 1 : today.date)
```
Comment (pmuchowski) [inline grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: I think you should consider using joda time here, with the `LocalDate` you don't need to reset the time and you will have the `plusMonths()` and `withMonthOfYear()` methods.
To get the name of the moth you can use the `DateTimeFormat.forPattern("MMM").withLocale(Locale.US)` to format the date which will give you the first 3 letters of a month.
To get the max number of days in current month you can use `date.dayOfMonth().getMaximumValue()`.
@jmiranda what do you think?

File: grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy [hunk 4]
```diff
@@ -11,20 +11,47 @@ import org.pih.warehouse.tablero.IndicatorDatasets
 
 class IndicatorDataService {
 
+    def dashboardService
     Date today = new Date()
```
[the remark this answers is not in the document]
Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: Thank you, @pmuchowski I noticed this as well a few PRs ago and forgot to comment. We need to create the `today` date within each closure / action.

File: grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy (hunk 3 above)
Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: @pmuchowski I'm was just confused in general about what a lot of the date data code is trying to accomplish so I've requested comments on everything, so I could recommend ways to improve it. But as a general rule, I agree, TimeCategory and JodaTime should be considered friends.

Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: And in general, err on the side of making the code readable.

File: src/js/components/tablero/GraphCard.jsx (hunk 2 above)
Comment (bamoure) [reply src/js/components/tablero/GraphCard.jsx]: This is for the time filter label to change between "Last x months" for indicators that show past data and "Next x months" for the Expiration Summary

File: grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy (hunk 4 above)
[the remark this answers is not in the document]
Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: @augusto-moraes We should remove the line `Date today = new Date()` as well.

**Pull request merged**